Library/Backends/ZendEngine2/Backend.php

```diff
@@ -269,12 +269,34 @@ public function onPostCompile($method, CompilationContext $context)
             $codePrinter->output('}');
             $codePrinter->decreaseLevel();
         }
+        else {
+            $cleanup = '';
+            foreach ($context->symbolTable->getVariables() as $variable) {
+                if ($variable->getName() != 'this_ptr' && $variable->getName() != 'return_value' && $variable->getName() != 'return_value_ptr') {
```

Comment:
This gathers all used variables (the code is from https://github.com/phalcon/zephir/blob/master/Library/ClassMethod.php#L1999), then retrieves only those which are initialized with `zval_used_for_init` (https://github.com/phalcon/zephir/blob/master/Library/Backends/ZendEngine2/Backend.php#L268).